include/gadget/usdt_argument.h

```diff
@@ -0,0 +1,360 @@
+// SPDX-License-Identifier: (LGPL-2.1 OR BSD-2-Clause)
+// Copyright (c) 2024 The Inspektor Gadget authors
+
+#ifndef __USDT_ARGUMENT_H
+#define __USDT_ARGUMENT_H
+
+#include <bpf/bpf_helpers.h>
+#include <bpf/bpf_core_read.h>
+#include <bpf/bpf_tracing.h>
+
+#define USDT_ARG_VERSION 1
+
+enum USDT_ARG_TYPE {
+	/* Argument in register */
+	USDT_ARG_TYPE_REG,
+	/* Argument in memory */
+	USDT_ARG_TYPE_MEM,
+};
+
+enum USDT_ARG_LENGTH {
```

Comment:
Did you design this enum yourself or is it part of some USDT spec/convention?

Does the ebpf program need to know whether it is signed or unsigned?

Reply:
I designed them by myself.

You are right, the signed/unsigned is not important for ebpf parts. I can remove them from the enum.

Reply:
The spec is here: https://sourceware.org/systemtap/wiki/UserSpaceProbeImplementation

For example, the length property just has 8 possible choices. But no one have tried to encode this, there are also no such enums in other projects.